### PR TITLE
Update basedocs.jl

### PR DESCRIPTION
**Pull request by litaxc**
Fix the typo in doc examples
in function map on line 557
and macro on line 164

### CODE REVIEW

File: base/docs/basedocs.jl [hunk 1]
```diff
@@ -161,7 +161,7 @@ end
     include expressions, literal values, and symbols. For example:
 
         macro sayhello(name)
-            return :( println("Hello, ", \$name) )
+            return :( println("Hello, ", $name) )
```
Comment (hayd) [inline base/docs/basedocs.jl]: This is inside a docstring, so this `$` needs to be escaped... If you do `help> macro` you'll see it correctly printing without the `\`. Did you see this render incorrectly somewhere?

The other change is good though.

Comment (litat) [inline base/docs/basedocs.jl]: Oh, I see.
I simply copy & paste to the Julia, found it generate an Error.
So should I undo the change on line 164 and fire another pull request?
Excuse me for the first time collaboration.

Comment (hayd) [inline base/docs/basedocs.jl]: Thanks for the contribution!

Either you can create a new PR, or force push to this branch (it'll update this PR).

**Comment by pranitbauva1997** [comment]
@hayd , @litat has made a commit. It is ready to be merged.

**Comment by tkelman** [comment]
squashed and pushed as 1380bfe231858b49f0395377ee0af1aa383bbf93 - thanks for catching this!
